services/app-api/src/postgres/contractAndRates/withdrawContract.ts

```diff
@@ -151,20 +156,33 @@ const withdrawContractInsideTransaction = async (
     })
 
     if (!rates) {
-        throw new NotFoundError('Could not find rates on contract')
+        throw new NotFoundError(
+            'Could not find rates on contract to be withdrawn'
+        )
     }
 
     // Child rates to withdraw with the submission, so it includes the same withdraw reason
     const ratesToWithdraw = []
+    // Child rates that need to be reassigned a new parent contract
+    const ratesToReAssign: {
+        rateID: string
+        rateName?: string
+        reassignToContractID: string
+        contractStatus: ConsolidatedContractStatus
+    }[] = []
 
-    // loop through all rates to determine which ones can be withdrawn.
+    // Loop through all rates to determine which ones can be withdrawn or reassigned parent contract
```

Comment:
suggestion:  think this long loop is a good candidate also for standalone function. Seems like it has has a type of something  like `prepareToWithdrawRates(rates) => [ratesToWithdraw, ratesToResassignParent]`